src/fu-tool.c

```diff
@@ -620,6 +623,22 @@ fu_util_update_device_changed_cb (FwupdClient *client,
 		priv->completion_flags |= FWUPD_DEVICE_FLAG_NEEDS_SHUTDOWN;
 	else if (fwupd_device_has_flag (device, FWUPD_DEVICE_FLAG_NEEDS_REBOOT))
 		priv->completion_flags |= FWUPD_DEVICE_FLAG_NEEDS_REBOOT;
+
+	if (priv->current_message == NULL) {
+		message = fwupd_device_get_update_message (priv->current_device);
+		if (message != NULL)
+			priv->current_message = g_strdup (message);
+	}
+}
+
+static void
+fu_util_display_current_message (FuUtilPrivate *priv)
+{
+	if (priv->current_message == NULL)
+		return;
+	g_print ("%s\n", priv->current_message);
+	g_free (priv->current_message);
```

Comment:
I think you can do `g_clear_pointer (&priv->current_message, g_free)` here to do it atomically.